wasm/module_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package wasm_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-interpreter/wagon/wasm"
+)
+
+func TestReadModule(t *testing.T) {
+	fnames, err := filepath.Glob(filepath.Join("testdata", "*.wasm"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, fname := range fnames {
+		testModule(t, fname)
+	}
+}
+
+func testModule(t *testing.T, fname string) {
+	f, err := os.Open(fname)
```

Comment:
Have you considered the impact this change makes on the test behavior?

If there is an error reading the file half-way through, previously, it would've been reported as an error reading the file, and `wasm.ReadModule` would not even be called.

But now, the error will come from `wasm.ReadModule`, since the `io.Reader` given to it may fail.

Both ways have tradeoffs. I just wanted to make sure this was an intentional change.

(P.S. I was leaving this comment on the 3rd commit, but you've squashed & force pushed while I was writing it, so I have to post it on the final diff.)